### DISCUSSION TITLE
Please add the ability to send messages by "Signal" or E-Mail

### DISCUSSION THREAD

**Discussion by berti34** [garethgeorge/backrest#395] [Ideas]
Hey George,
thank you for this great software! I've been looking for such active project since the beginning of restic.

What do you think about adding a hook feature or "plugin" to send messages/notifications by Signal, Zulip, or even E-Mail natively, especially on windows?

Or, does somebody know such small cli software tools to use as hooks on windows?

Thanks a lot!
Berthold

**Comment by berti34** [comment]
I've found this well working tool for Windows:
  https://github.com/KeepSec-Technologies/Mail2Go
which is based on
  https://github.com/emersion/go-smtp

I've to add two newlines after the hook cmdline, to get it to run:
c:\mail2go\mail2go.exe -c c:\mail2go\config.json -t incoming@example.com -h "Backrest on LAP-1, Event: {{ .EventName .Event }}" -b {{ .ShellEscape .Summary }}-NL-
-NL-

Maybe this helps somebody.

**Reply by SebastianDonhauser** [reply]
Thank you for the advice. I'm already using mail2go in another context and hadn't thought of using it here. E-Mail is and remains the most flexible solution.

**Comment by Chucky2401** [comment]
I use the container version of apprise (https://hub.docker.com/r/caronc/apprise) on my side.
Easy to setup, and just use one URL to send your message.

**Reply by clockwinder** [reply]
This is the solution I'm looking to set up.

Would you be willing to post your hook (assuming that's how you're accomplishing this)?  It would be a great reference/starting point for me!

**Reply by Chucky2401** [reply]
Sure!

Here it is:

```shell
curl -X POST -F "body={{ .Summary }}" http://x.x.x.x:8010/notify/backrest
```

`backrest` is the configuration key from Apprise API container.

Apprise is very-well documented, and in the overview of your apprise api, you have some examples on how to use it.

If you want more information, feel free to ask me.
